### PR TITLE
Significant Module Refactor

### PR DESCRIPTION
**Pull request by lyzadanger**
Ahhhhhh....sooooooo much cleaner. Modules are broken down and clearer and there aren't modules exporting a whole bunch of random functions (that's left for `utils` modules).

The `drizzle` flow is, generally:

1. `prepare` - register helpers, partials and other template preparation
2. `parse` - parse content and data from different kinds of source files and build context objects
3. `render` - compile templates
4. `write` - write files out to the filesystem

Module files are arranged according to that now. Far less oddly-coupled things.

Though structure has changed quite a bit here, functionality hasn't much. I've just moved things around. I'm going to spend the rest of the day writing a bunch more tests to fill in some gaps...

/cc @mrgerardorodriguez

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -8,13 +10,7 @@ import prepare from './prepare';
...
 function drizzle (options) {
   const opts = parseOptions(options);
-  return prepare(opts).then(drizzleData => {
-    return {
-      context: drizzleData.context,
-      handlebars: drizzleData.handlebars,
-      options: opts
-    };
-  });
+  return prepare(opts).then(parse).then(write);
```
Comment (lyzadanger) [inline src/index.js]: Ahhhhh that feels better :)

File: src/write/pages.js [hunk 2]
```diff
@@ -46,9 +40,9 @@ function writePages (pages, drizzleData, writePromises = []) {
  * TODO: Better resolve/return value
  */
 function pages (drizzleData) {
-  return Promise.all(writePages(drizzleData.context.pages, drizzleData))
+  return Promise.all(writePages(drizzleData.pages, drizzleData))
     .then(() => {
-      return true;
+      return drizzleData;
```
Comment (erikjung) [inline src/write/pages.js]: So to be clear, the promises mutate `drizzleData` and then we return the resulting, changed object?

File: src/prepare/templates.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * @param {Object} opts Drizzle options
+ * @return {Promise} resolves to {Object} Handlebars instance
+ */
+function prepareTemplates (opts) {
+  return Promise.all([
+    prepareHelpers(opts.handlebars, opts.helpers),
+    preparePartials(opts.handlebars, opts.partials)
+  ]).then(handlebarsInfo => {
+    return opts;
+  });
```
Comment (mrgerardorodriguez) [inline src/prepare/templates.js]: Could this be simplified?

`.then( handlebarsInfo => opts )`

Another thought, is `handlebarsInfo` needed? It seems it isn't used?

**Comment by erikjung** [comment]
@lyzadanger I had one small question, but I think :+1:

File: src/write/pages.js (hunk 2 above)
Comment (lyzadanger) [inline src/write/pages.js]: I don't believe it mutates it, actually; it's returning it for chaining reasons.

File: src/prepare/templates.js (hunk 3 above)
Comment (lyzadanger) [inline src/prepare/templates.js]: This is in a sort of half-done state here...I need to refactor this a touch.

Comment (mrgerardorodriguez) [inline src/prepare/templates.js]: Got it. :)

**Comment by mrgerardorodriguez** [comment]
This is looking great @lyzadanger! Well done! :grinning:

**Pull request merged**